### PR TITLE
Add --set flag

### PR DESCRIPTION
**Pull request by autrilla**
The behavior is a bit different from the Python version. Instead of `--set '["app2"]["key"]' '"app2keystringvalue"'`, you have to do `--set '["app2"]["key"] "app2keystringvalue"'`

### CODE REVIEW

File: cmd/sops/main.go [hunk 1]
```diff
@@ -192,6 +195,10 @@ func main() {
 			Name:  "encryption-context",
 			Usage: "comma separated list of KMS encryption context key:value pairs",
 		},
+		cli.StringFlag{
+			Name:  "set",
+			Usage: `set a specific key or branch in the input JSON or YAML document. value must be a json encoded string. (edit mode only). eg. --set '["somekey"][0] {"somevalue":true}'`,
```
Comment (jvehent) [inline cmd/sops/main.go]: Does `--set` open an editor? I thought it would just set the value and exit.

File: cmd/sops/main.go [hunk 2]
```diff
@@ -557,77 +564,115 @@ func edit(c *cli.Context, file string, fileBytes []byte) ([]byte, error) {
...
 		}
-		if bytes.Equal(newHash, origHash) {
-			return nil, cli.NewExitError("File has not changed, exiting.", exitFileHasNotBeenModified)
+		branch := parent.(sops.TreeBranch)
+		for i, kv := range branch {
+			if kv.Key == key {
+				branch[i].Value = valueTree
+				break
+			}
 		}
```
Comment (jvehent) [inline cmd/sops/main.go]: Would it be cleaner to move the above chunk of code to a function outside of `main()` ?

File: cmd/sops/main.go [hunk 3]
```diff
@@ -557,77 +564,115 @@ func edit(c *cli.Context, file string, fileBytes []byte) ([]byte, error) {
...
-		return nil, cli.NewExitError(fmt.Sprintf("Could not write output file: %s", err), exitCouldNotWriteOutputFile)
-	}
-	origHash, err := hashFile(tmpfile.Name())
-	if err != nil {
-		return nil, cli.NewExitError(fmt.Sprintf("Could not hash file: %s", err), exitCouldNotReadInputFile)
-	}
-	for {
-		err = runEditor(tmpfile.Name())
+	if c.String("set") != "" {
+		pathValuePair := strings.SplitAfterN(c.String("set"), "] ", 2)
```
Comment (jvehent) [inline cmd/sops/main.go]: Need a comment that describes the algorithm we're following here, this is too convoluted to just follow the code.

File: cmd/sops/main.go (hunk 1 above)
Comment (autrilla) [inline cmd/sops/main.go]: It doesn't, why are you asking?

Comment (jvehent) [inline cmd/sops/main.go]: Before the `Usage` string says `(edit mode only)`.

Comment (autrilla) [inline cmd/sops/main.go]: Oh, I just copied that from the python version https://github.com/mozilla/sops/blob/python-sops/sops/__init__.py#L148-L153

**Pull request merged**